### PR TITLE
Instance cleanup

### PR DESCRIPTION
**Pull request by akallabeth**
* Unifies and cleans up instance creation/destruction
* Deprecates duplicate fields in `freerdp* instance`

### CODE REVIEW

File: server/proxy/pf_client.c [hunk 1]
```diff
@@ -619,9 +620,9 @@ static BOOL pf_client_post_connect(freerdp* instance)
 	WINPR_ASSERT(instance);
 	context = instance->context;
 	WINPR_ASSERT(context);
-	settings = instance->settings;
+	settings = instance->context->settings;
```
Comment (mfleisz) [inline server/proxy/pf_client.c:623]: `context->settings` is enough

**Review by mfleisz — COMMENTED** [review]
A few comments but overall LGTM - nice cleanup

File: client/Mac/MRDPView.m [hunk 2]
```diff
@@ -1377,7 +1423,7 @@ void sync_keyboard_state(freerdp *instance)
 		if (currentFlags & kCGEventFlagMaskNumericPad)
 			flags |= KBD_SYNC_NUM_LOCK;
 
-		freerdp_input_send_synchronize_event(instance->input, flags);
+		freerdp_input_send_synchronize_event(instance->context->input, flags);
```
[the remark this answers is not in the document]
Comment (akallabeth) [reply client/Mac/MRDPView.m:1426]: @mfleisz no, `context` is of type `mfContext`

File: client/Wayland/wlfreerdp.c [hunk 3]
```diff
@@ -204,7 +204,7 @@ static BOOL wl_pre_connect(freerdp* instance)
 		}
 	}
 
-	if (!freerdp_client_load_addins(instance->context->channels, instance->settings))
+	if (!freerdp_client_load_addins(instance->context->channels, settings))
```
[the remark this answers is not in the document]
Comment (akallabeth) [reply client/Wayland/wlfreerdp.c:207]: no, `context` is of type `wlfContext`

File: client/Wayland/wlfreerdp.c [hunk 4]
```diff
@@ -261,8 +261,8 @@ static BOOL wl_post_connect(freerdp* instance)
 	UwacWindowSetTitle(window, title);
 	UwacWindowSetAppId(window, app_id);
 	UwacWindowSetOpaqueRegion(context->window, 0, 0, w, h);
-	instance->update->EndPaint = wl_end_paint;
-	instance->update->DesktopResize = wl_resize_display;
+	instance->context->update->EndPaint = wl_end_paint;
+	instance->context->update->DesktopResize = wl_resize_display;
```
[the remark this answers is not in the document]
Comment (akallabeth) [reply client/Wayland/wlfreerdp.c:265]: no, `context` is of type `wlfContext`

**Review by mfleisz — APPROVED** [review]

**Comment by vp1981** [comment]
Hello.
The changes to `xf_client.c` has one drawback: the line 1578 uses unassigned variable, I think you missed variable assignment:
```
diff --git a/client/X11/xf_client.c b/client/X11/xf_client.c
index 695661fdd..57a8187b5 100644
--- a/client/X11/xf_client.c
+++ b/client/X11/xf_client.c
@@ -1574,6 +1574,7 @@ static DWORD WINAPI xf_client_thread(LPVOID param)
        if (!status)
                goto end;

+       settings = context->settings;
        /* --authonly ? */
        if (settings->AuthenticationOnly)
        {
@@ -1594,7 +1595,6 @@ static DWORD WINAPI xf_client_thread(LPVOID param)
                goto disconnect;
        }

-       settings = context->settings;
        timer = CreateWaitableTimerA(NULL, FALSE, "mainloop-periodic-timer");

        if (!timer)
```

Without that patch the xfreerdp segfaults.

**Comment by akallabeth** [comment]
@vp1981 thank you for the hint, already fixed in #7740

**Pull request merged**